example/example/tests/test_customer_views.py

```diff
@@ -366,3 +366,65 @@ def test_delete_customer_user_sends_email(self, admin, user, mock_storage):
         email = mail.outbox[0]
         assert len(mail.outbox) == 1
         assert user.email in email.to
+
+    def test_resend_invitation_new_customer_user(self, admin, user_data, mock_storage):
+
+        customer = CustomerFactory(logo=None)
+
+        _, key = create_auth_token(admin)
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION=f"Token {key}")
+
+        url = reverse("customer-users-list", args=[customer.id])
+        data = {
+            "customer": customer.name,
+            "user": user_data,
+        }
+        response = client.post(url, data, format="json")
+        assert status.is_success(response.status_code)
+        old_content = response.json()
+
+        assert old_content["invitation_date"] is not None
+        assert old_content["user"]["change_password"] is True
+
+        url = reverse("customer-users-invite", args=[customer.id, old_content["user"]["id"]])
+        response = client.post(url, {}, format="json")
+        assert status.is_success(response.status_code)
+        new_content = response.json()
+
+        assert new_content["invitation_date"] != old_content["invitation_date"]
+        assert new_content["invitation_date"] is not None
+        assert new_content["user"]["change_password"] is True
+
+        RESET_PASSWORD_TEXT = "Please follow the link below to create a User Account and Password"
+
+        assert len(mail.outbox) == 2
```

Comment:
So the code to invite a user, sends 2 emails? that is the first I'm aware of this. Not against it, just didn't get that from the above code. Where in the code above does this happen?

I have just re-read the whole file and it looks like a previous test is inviting a user, this is about re-sending the invite. Does this mean that tests bleed into each other and change some shared state? We don't do that in Java/JavaScript. All tests should be isolated, I'm more asking so I am aware of this, rather than asking for it to be changed (if I'm correct), but I do think it is a bad idea, especially for newbs like me.....